tasks/kernel_matrix_testing/compiler.py

```diff
@@ -179,48 +168,83 @@ def start(self) -> None:
-            self.prepare_for_cross_compile()
-
-    def prepare_for_cross_compile(self):
-        target = ARCH_AMD64 if self.arch == ARCH_ARM64 else ARCH_ARM64
-
-        # Hardcoded links to the header packages for each architecture. Why do this and not have something more automated?
-        # 1. While right now the URLs are similar and we'd only need a single link with variable replacement, this might
-        #    change as the repository layout is not under our control.
-        # 2. Automatic detection of these URLs is not direct (querying the package repo APIs is not trivial) and we'd need some
-        #    level of hard-coding some URLs or assumptions anyways.
-        # 3. Even if someone forgets to update these URLs, it's not a big deal, as we're building inside of a Docker image which will
-        #    likely have a different kernel than the target system where the built eBPF files are going to run anyways.
-        header_package_urls: dict[Arch, str] = {
-            ARCH_AMD64: AMD64_DEBIAN_KERNEL_HEADERS_URL,
-            ARCH_ARM64: ARM64_DEBIAN_KERNEL_HEADERS_URL,
-        }
-
-        header_package_path = "/tmp/headers.deb"
-        self.exec(f"wget -O {header_package_path} {header_package_urls[target]}")
+        res = self.exec(
+            f"apt-get download {headers_package}:{cross_arch.go_arch} --print-uris | awk '{{ print \\$2 }}'",
+            user="root",
+        )
+        headers_package_filename = res.stdout.strip()
+        self.exec(f"apt-get download {headers_package}:{cross_arch.go_arch}", user="root")
 
         # Uncompress the package in the root directory, so that we have access to the headers
         # We cannot install because the architecture will not match
         # Extract into a .tar file and then use tar to extract the contents to avoid issues
         # with dpkg-deb not respecting symlinks.
-        self.exec(f"dpkg-deb --fsys-tarfile {header_package_path} > {header_package_path}.tar", user="root")
-        self.exec(f"tar -h -xf {header_package_path}.tar -C /", user="root")
+        self.exec(f"dpkg-deb --fsys-tarfile {headers_package_filename} > {headers_package_filename}.tar", user="root")
+        self.exec(f"tar --skip-old-files -xf {headers_package_filename}.tar -C /", user="root")
+        self.exec(
+            f"mv /usr/src/{headers_package}/include/generated/*.h /usr/src/{headers_package}/arch/{cross_arch.kernel_arch}/include/generated/",
+            user="root",
+        )
+
+        self.exec("apt-get install -y --no-install-recommends sudo", user="root")
+        self.exec("usermod -aG sudo compiler && echo 'compiler ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers", user="root")
+        self.exec(f"cp /root/.bashrc /home/compiler/.bashrc && chown {uid}:{gid} /home/compiler/.bashrc", user="root")
+        self.exec("mkdir ~/.cargo && touch ~/.cargo/env", user="compiler")
+        self.exec("dda self telemetry disable", user="compiler", force_color=False)
```

Comment:
Would an environment variable be useful to prevent having to do this?

Reply:
Yes, it would.